server/src/main/java/com/cloud/network/guru/DirectNetworkGuru.java

```diff
@@ -367,9 +358,6 @@ public void doInTransactionWithoutResult(TransactionStatus status) {
             }
         }
 
-        if (nic.getIPv6Address() != null) {
-            _ipv6Mgr.revokeDirectIpv6Address(nic.getNetworkId(), nic.getIPv6Address());
```

Comment:
@wido I did not understand this part, who would revoke ipv6 address? Or is the assumption that it is not needed.

Reply:
@rhtyd In the IPv4 world we would have a pool of addresses which we would use. These would need to be Allocated and Free'd again once no longer in use.

With IPv6 and SLAAC the Instance generates a IPv6 address based on it's MAC address.

Therefor we do not have a pool of IPs which we need to keep on the Management Server side. We just generate the EUI-64 address based on the subnet+mac and store that at the NIC.

If we clear the NIC the address is 'Free' again, but any new Instance will have a new MAC and thus a new IP.

SLAAC: https://howdoesinternetwork.com/2013/slaac